contracts/proxy/diamond/writable/DiamondWritableInternal.sol

```diff
@@ -148,12 +148,17 @@ abstract contract DiamondWritableInternal is IDiamondWritableInternal {
                 if (address(bytes20(oldFacet)) == address(this))
                     revert DiamondWritable__SelectorIsImmutable();
 
-                if (selectorSlot == 0) {
+                // selectorSlot for 0x00000000 selector == 0
+                // We also need the below to trigger for 0x00000000 selector when selectorInSlotIndex !=0
+                if (
+                    selectorSlot != 0 ||
+                    (selectorInSlotIndex != 0 && selector == 0)
+                ) {
+                    selectorInSlotIndex--;
+                } else {
                     selectorSlotCount--;
                     selectorSlot = l.selectorSlots[selectorSlotCount];
                     selectorInSlotIndex = 7;
```

Comment:
Invert the logic and revert the order of the if/else blocks:
```suggestion
                if (
                    selectorSlot == 0 &&
                    (selectorInSlotIndex == 0 || selector != 0)
                ) {
                    selectorSlotCount--;
                    selectorSlot = l.selectorSlots[selectorSlotCount];
                    selectorInSlotIndex = 7;
                } else {
                    selectorInSlotIndex--;
                    
```

Reply:
Also I'm rather confident that the `selector` comparison is redundant.  And I'm somewhat confident that `selectorSlot` comparison is also redundant.  Isn't the `selectorInSlotIndex` comparison all we need?  Tests pass if I remove the others.

Reply:
Yeah, I think you are definitely right about this. Updating.